apps/docs/pages/guides/integrations/prisma.mdx

```diff
@@ -115,53 +112,83 @@ model User {
-Go to the **Database** page from the sidebar in the Supabase dashboard and navigate to **connection pool** settings
+If you’re working in a serverless environment (for example Node.js functions hosted on AWS Lambda, Vercel or Netlify Functions), you need to set up [connection pooling](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#serverless-environments-faas) using a tool like [PgBouncer](https://www.pgbouncer.org/). That’s because every function invocation may result in a [new connection to the database](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#the-serverless-challenge). 
+
+Supabase [supports connection management using PgBouncer](/docs/guides/database/connecting-to-postgres#connection-pool).
+
+Go to the **Database** page from the sidebar in the Supabase dashboard and navigate to **Connection pool** settings:
+
 ![Connection pool settings](/docs/img/guides/integrations/prisma/w0oowg8vq435ob5c3gf0.png)
-When migrating, you need to use the non-pooled connection URL (like the one used in **step 1**). However, when deploying your app, use the pooled connection URL and add the `?pgbouncer=true` flag to the PostgreSQL connection URL. It's also recommended to minimize the number of concurrent connections by setting the `connection_limit` to `1`. The `.env` file should look like:
+
+When updating your database schema, you need to use the non-pooled connection URL (like the one used in **step 1**). You can configure the non-pooled connection string by using the `directUrl` property in the datasource block.
+
+Update your `.env` file with the following changes:
+1. Rename the `DATABASE_URL` environment variable to `DIRECT_URL`
+1. Create a `DATABASE_URL` environment variable and paste in the new connection string from the dashboard as its value
+
+It is recommended to minimize the number of concurrent connections by setting the `connection_limit` to `1`. You can set connection limit by appending `?connection_limit=1` to your connection string
+
+Your `.env` file should resemble the following:
 
 ```env
 # .env
-DATABASE_URL="postgres://postgres:[YOUR-PASSWORD]@db.[YOUR-PROJECT-REF].supabase.co:6543/postgres?pgbouncer=true&connection_limit=1"
+# PostgreSQL connection string used for migrations
+DIRECT_URL="postgres://postgres:[YOUR-PASSWORD]@db.[YOUR-PROJECT-REF].supabase.co:6543/postgres"
+# PostgreSQL connection string with pgBouncer config — used by Prisma Client
+DATABASE_URL="postgres://postgres:[YOUR-PASSWORD]@db.[YOUR-PROJECT-REF].supabase.co:6543/postgres?connection_limit=1"
```

Comment:
This needs `pgbouncer=true`: https://www.prisma.io/docs/guides/performance-and-optimization/connection-management/configure-pg-bouncer

Reply:
From an internal conversation

> I think the `pgbouncer=true` was lifted from your docs — not needed on our end, as they connect to an entirely different port that only hosts pgbouncer.
`connection_limit=1` - I believe this tells prisma to not spin up multiple connections? We added this a while ago for folks that were using prisma from serverless envs (e.g. Vercel functions). I don’t remember the defaults that were being used, but each invocation (== page load, for example) would spin up 10 connections, quickly depleting pgbouncer’s available connections

Reply:
Prisma requires `pgbouncer=true` to be able to talk to a PgBouncer instead of a direct database. There is a reason why we document it as required.